### PR TITLE
Update/my subscriptions wp installer

### PR DESCRIPTION
**Pull request by bgrgicak**
### Changes proposed in this Pull Request:

This PR is part of the My subscriptions feature branch.

It adds support for using the WordPress installer in My subscriptions.
For the installer to work this PR had to add download paths to the info request for themes and plugins.

Because there isn't a WordPress API endpoint for activating plugins and themes, we had to add one to the marketplace API.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the [My subscriptions page](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions) 
2. Connect to local Woo.com (branch `add/product-slug-to-helper-subscriptions`)
3. Install a WCCOM extensions, .org extension, and a theme
4. Confirm that installing works

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

Use WP installer for my subscriptions instead of the Woo installer API.

</details>

https://github.com/woocommerce/woocommerce/assets/1199991/b7399709-ba08-4c9b-b1b1-0ef62d226925

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/utils/functions.tsx [hunk 1]
```diff
@@ -199,17 +203,114 @@ function installProduct( productKey: string ): Promise< void > {
...
+		},
+		body: data,
+	} );
+}
+
+function wpAjax(
+	action: string,
+	data: {
+		slug?: string;
+		plugin?: string;
```
Comment (mlaetitia) [inline plugins/woocommerce-admin/client/marketplace/utils/functions.tsx:227]: I am not seeing where these plugin and theme are being used. Could you point me to it ?

File: plugins/woocommerce/includes/admin/helper/class-wc-helper-plugin-info.php [hunk 2]
```diff
@@ -74,6 +99,11 @@ public static function plugins_api( $response, $action, $args ) {
 		$results = json_decode( wp_remote_retrieve_body( $request ), true );
 		if ( ! empty( $results ) ) {
 			$response = (object) $results;
+
+			$product = array_shift( $products );
+			if ( ! is_null( $product ) && isset( $product['package'] ) ) {
```
Comment (mlaetitia) [inline plugins/woocommerce/includes/admin/helper/class-wc-helper-plugin-info.php]: Do we need both conditions ? If $product is null,` isset( $product['package'] )` will return false anyway. Is this for readability of code ?

```suggestion
			if ( isset( $product['package'] ) ) {
```

**Review by mlaetitia — APPROVED** [review]
Tested and it works as expected! Tested for themes, extensions and wporg extensions.

Still unsure about the behavior for SaaS products (if we have a SaaS product without a zip file - we can click Install, which will connect the product to the site, but nothing will be installed, cause there's nothing to install).

Left a couple of comments throughout, but they aren't blockers.

**Comment by bgrgicak** [comment]
> Still unsure about the behavior for SaaS products (if we have a SaaS product without a zip file - we can click Install, which will connect the product to the site, but nothing will be installed, cause there's nothing to install).

We should exclude these products from the API.

File: plugins/woocommerce-admin/client/marketplace/utils/functions.tsx (hunk 1 above)
Comment (bgrgicak) [reply plugins/woocommerce-admin/client/marketplace/utils/functions.tsx:227]: It's part of WP https://github.com/WordPress/WordPress/blob/master/wp-admin/update.php#L23C6-L24C3

File: plugins/woocommerce/includes/admin/helper/class-wc-helper-plugin-info.php (hunk 2 above)
Comment (bgrgicak) [reply plugins/woocommerce/includes/admin/helper/class-wc-helper-plugin-info.php]: No, I was just being to defensive. Removed `is_null` in [026b899](https://github.com/woocommerce/woocommerce/pull/40990/commits/026b8992fdc6e92c45b4dfbeee37cafbc21409a4).

File: plugins/woocommerce-admin/client/marketplace/utils/functions.tsx (hunk 1 above)
Comment (mlaetitia) [reply plugins/woocommerce-admin/client/marketplace/utils/functions.tsx:227]: AH got it ! Nevermind then!

**Review by raicem — APPROVED** [review]
Worked great, thanks @bgrgicak!

**Pull request merged**